### ISSUE TITLE
Incorrect compiler error message because of wrong code parsing

### ISSUE THREAD

**Issue by lilianmoraru** [rust-lang/rust#31023]
I was modifying [this code](http://benchmarksgame.alioth.debian.org/u64q/program.php?test=mandelbrot&lang=rust&id=1) and I've got this error message:
```
   Compiling mandelbrot v0.1.0 (file:///home/uidr1070/RustProjects/mandelbrot)
src/main.rs:78:17: 80:33 error: the trait `core::ops::Div<()>` is not implemented for the type `usize` [E0277]
src/main.rs:78     let split = size / THREADS/
src/main.rs:79 
src/main.rs:80     assert!(size % THREADS == 0);// FIXME
src/main.rs:78:17: 80:33 help: run `rustc --explain E0277` to see a detailed explanation
error: aborting due to previous error
Could not compile `mandelbrot`.
```

Instead of putting the semicolon(`;`) at the end of `let split = size / THREADS`, I pressed `/`. I guess it tries to divide the assert.
Probably hard to parse this... but one would expect to see this message:
```
error: expected one of `!`, `.`, `::`, `;`, `{`, or an operator
```
Not a big issue, just informing...

**Comment by oli-obk** [comment]
It could give a `help` message suggesting a semicolon, when encountering an unimplemented binary operation where the rhs is a `()`.

**Comment by arielb1** [comment]
The error message looks "correct" to me.

**Comment by lilianmoraru** [comment]
@arielb1 Ye, only a human understands what you intended to do(some might actually think that you tried a multi-line division).
For a compiler this is just a multi-line division, so it is definitely correct from this point of view.
